llvm/test/Analysis/ScalarEvolution/infer-trip-count.ll

```diff
@@ -0,0 +1,191 @@
+; NOTE: Assertions have been autogenerated by utils/update_analyze_test_checks.py
```

Comment:
The tests need to be improved by adding:
1. Negative step
2. Multi-dim array with nested loop with both forwarded and reversed indices